README.md
[![image](https://img.shields.io/travis/seisman/HinetPy/master.svg)](https://travis-ci.org/seisman/HinetPy)
[![image](https://img.shields.io/github/release/seisman/HinetPy.svg)](https://github.com/seisman/HinetPy/releases)
[![image](https://img.shields.io/pypi/v/HinetPy.svg)](https://pypi.python.org/pypi/HinetPy/)
[![image](https://img.shields.io/github/license/seisman/HinetPy.svg)](https://github.com/seisman/HinetPy/blob/master/LICENSE)
[![image](https://img.shields.io/pypi/pyversions/HinetPy.svg)](https://pypi.python.org/pypi/HinetPy/)

[HinetPy](https://seisman.github.io/HinetPy) is a Python package aiming
to automate and simplify tedious data request, downloading and format
conversion tasks related to [NIED Hi-net](http://www.hinet.bosai.go.jp/).

Features
========

1.  Automatically request continuous waveform data from Hi-net
2.  Convert win32 data into SAC format
3.  Extract instrumental response as SAC PZ file

A simple example
================

It's simple to request a continuous waveform data from Hi-net, convert
the data into SAC format and extract instrumental response as SAC PZ
file.

```python
>>> from HinetPy import Client, win32
>>> from datetime import datetime
>>>
>>> # You need a Hi-net account to access their data
>>> client = Client("username", "password")
>>>
>>> # Let's try to request 20 minutes data since 2010-01-01T00:00(GMT+0900) from Hi-net
>>> starttime = datetime(2010, 1, 1, 0, 0)
>>> data, ctable = client.get_waveform('0101', starttime, 20)
>>> # The request process takes several minutes due to the unfriendly design of Hi-net
>>> ls  # the downloaded data and corresponding channel table
0101_201001010000_20.cnt 0101_20100101.ch
>>>
>>> # Let's convert win32 data into SAC format
>>> sacfiles = win32.extract_sac(data, ctable)
>>> ls *.SAC
N.NGUH.E.SAC  N.NGUH.U.SAC  N.NNMH.N.SAC
N.NGUH.N.SAC  N.NNMH.E.SAC  N.NNMH.U.SAC
...
>>>
>>> # Let's extract instrument response as PZ file from channel table
>>> pzfiles = win32.extract_pz(ctable)
>>> ls *.SAC_PZ
N.NGUH.E.SAC_PZ  N.NGUH.U.SAC_PZ  N.NNMH.N.SAC_PZ
N.NGUH.N.SAC_PZ  N.NNMH.E.SAC_PZ  N.NNMH.U.SAC_PZ
...
```

License
=======

This project is licensed under the terms of the MIT license.


Changelog.rst
Changelog
=========

0.6.5 (2019-12-06):
 - Fix PZ files if dampling constant is zero (#24)

0.6.4 (2019-08-23):
 - Capture exception caused by incorrect channel information (#22)

0.6.3 (2019-06-13):
 - Fix ``select_stations()`` (#19)
 - Rename ``string2datetime()`` to ``to_datetime()``

0.6.2 (2019-05-06):
 - Fix download focal mechanism catalog issue (#18).

0.6.1 (2019-02-20):
 - Fix selecting events in a rectangular region.

0.6.0 (2019-02-19):
 - Support request event waveform data (#16).
 - ``get_waveform()`` is renamed to ``get_continuous_waveform()``.

0.5.2 (2019-02-19):
 - Fix selecting stations in a rectangular region (#17).

0.5.1 (2018-12-08):
 - Fix typo from longtitude to longitude.

0.5.0 (2018-11-21):
 - Fix issues of wrong CONSTANT in SAC polezero file (#8).
 - Fix login failure issue with password longer than 12 characters (#13).

0.4.8 (2018-10-04):
 - ``get_station_list()``: must specify a network code; support S-net and MeSO-net.
 - ``select_stations()``: support S-net and MeSO-net

0.4.7 (2018-10-04):
 - Support S-net and MeSO-net (#9 and #10)
 - Fix an issue when channel table contains blank lines

0.4.6 (2018-03-20):
 - Fix ``Too many open files`` (#6)

0.4.5 (2018-03-07):
 - ``get_station_list()``: return a list of stations
 - ``select_stations()``: support selecting stations in a box or circular region

0.4.4 (2017-11-30):
 - Fix a technical issue related to packaging

0.4.3 (2017-11-30):
 - Add Chinese documentation

0.4.2 (2017-06-18):
 - Fix a bug with requests>=2.17

0.4.1 (2017-06-18):
 - remove tempfile after downloading.

0.4.0 (2017-04-01):
 - ``win32.extract_sac()``: skip if data not exists
 - ``win32.extract_sac()``: support multiple processes to speedup, and
   no longer return values
 - ``Client.get_waveform()``: support multi-threads to speedup
 - Change ``Client.help()`` to ``Client.info()``
 - ``Client.get_waveform()`` now can automatically set ``max_span``
 - ``Client.get_*()`` now support startime in different string formats

0.3.3 (2017-03-17):
 - Change ``network`` to ``NETWORK`` in ``header.py``
 - Add wildcard support to ``win32.merge()``
 - Change ``Client.check_module_release()`` to ``Client.check_package_release()``
 - Support output filename with deep directory
 - Always sort cnt files to avoid merge error
 - Set ``pmax`` to 8640000 by default
 - Fix typos

0.3.2 (2017-03-12):
 - Fix another technical issue related to pypi

0.3.1 (2017-03-12):
 - Fix a technical issue related to pypi

0.3.0 (2017-03-12):
 - Rewritten as a Python package

0.2.0 (2016-08-24):
 - Some small fixes and improvements

0.1.0 (2016-08-04):
 - First public release
 - ``HinetDoctor.py``: check dependencies
 - ``HinetContRequest.py``: request continuous data from Hi-net
 - ``StationSelector.py``: select Hi-net/F-net stations before requesting data
 - ``HinetJMARequest.py``: request JMA catalogs from Hi-net website
 - ``rdhinet.py``: convert WIN32 format to SAC format
 - ``ch2pz.py``: extract SAC PZ files from Hi-net channel table files


HinetPy/header.py
# -*- coding: utf-8 -*-
"""Basic information of networks."""

from datetime import datetime
from collections import namedtuple

Network = namedtuple("Network", "name, channels, starttime, url")
""" An object containing information of a network.

.. py:attribute:: name

   Network name.

.. py:attribute:: channels

   Number of channels the network has.

.. py:attribute:: starttime

   Start time (JST: UTC+0900) when waveform data is avaiable.

.. py:attribute:: url

   Homepage of the network.

``NETWORK`` is a dict of :class:`~HinetPy.header.Network`, containing
information of all networks available from Hi-net website.

>>> from HinetPy.header import NETWORK
>>> for code in NETWORK.keys():
...     print(code, NETWORK[code].name)
0101 NIED Hi-net
0103 NIED F-net (broadband)
0103A NIED F-net (strong motion)
0120 NIED S-net (speed motion)
0120A NIED S-net (acceleration)
0131 NIED MeSO-net
0201 Hokkaido University
...
"""

URL_Hinet = "http://www.hinet.bosai.go.jp"
URL_Fnet = "http://www.fnet.bosai.go.jp"
URL_Vnet = "http://www.vnet.bosai.go.jp"
URL_JMA = "http://www.jma.go.jp/jma/indexe.html"

NETWORK = {}

NETWORK["0101"] = Network(
    name="NIED Hi-net",
    channels=2336,
    starttime=datetime(2004, 4, 1, 0, 0),
    url=URL_Hinet,
)
NETWORK["0103"] = Network(
    name="NIED F-net (broadband)",
    channels=438,
    starttime=datetime(2004, 4, 1, 0, 0),
    url=URL_Fnet,
)
NETWORK["0103A"] = Network(
    name="NIED F-net (strong motion)",
    channels=438,
    starttime=datetime(2004, 4, 1, 0, 0),
    url=URL_Fnet,
)
NETWORK["0120"] = Network(
    name="NIED S-net (speed motion)",
    channels=450 * 2,  # 450 velocity + 450 acceleration
    starttime=datetime(2016, 8, 15, 0, 0),
    url=URL_Hinet,
)
NETWORK["0120A"] = Network(
    name="NIED S-net (acceleration)",
    channels=450 * 2,  # 450 velocity + 450 acceleration
    starttime=datetime(2016, 8, 15, 0, 0),
    url=URL_Hinet,
)
NETWORK["0131"] = Network(
    name="NIED MeSO-net",
    channels=900,
    starttime=datetime(2017, 4, 1, 0, 0),
    url=URL_Hinet,
)

NETWORK["0201"] = Network(
    name="Hokkaido University",
    channels=183,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.sci.hokudai.ac.jp/isv/english/",
)
NETWORK["0202"] = Network(
    name="Tohoku University",
    channels=157,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.aob.geophys.tohoku.ac.jp/aob-e/",
)
NETWORK["0203"] = Network(
    name="Tokyo University",
    channels=301,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.eri.u-tokyo.ac.jp/eng/",
)
NETWORK["0204"] = Network(
    name="Kyoto University",
    channels=196,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.dpri.kyoto-u.ac.jp/en/",
)
NETWORK["0205"] = Network(
    name="Kyushu University",
    channels=73,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.sevo.kyushu-u.ac.jp/index-e.html",
)
NETWORK["0206"] = Network(
    name="Hirosaki University",
    channels=13,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://hrsryu.geo.hirosaki-u.ac.jp/",
)
NETWORK["0207"] = Network(
    name="Nagoya University",
    channels=75,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.seis.nagoya-u.ac.jp/index_e/",
)
NETWORK["0208"] = Network(
    name="Kochi University",
    channels=28,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www-en.kochi-u.ac.jp/",
)
NETWORK["0209"] = Network(
    name="Kagoshima University",
    channels=45,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://leopard.sci.kagoshima-u.ac.jp/noev/English/home.htm",
)

NETWORK["0301"] = Network(
    name="JMA Seismometer Network",
    channels=869,
    starttime=datetime(2004, 4, 1, 0, 0),
    url=URL_JMA,
)

NETWORK["0401"] = Network(
    name="JAMSTEC Realtime Data " "from the Deep Sea Floor Observatory",
    channels=73,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.jamstec.go.jp/e/index.html",
)
NETWORK["0402"] = Network(
    name="NIED DONET1 (broadband)",
    channels=132,
    starttime=datetime(2014, 10, 1, 0, 0),
    url=URL_Hinet,
)
NETWORK["0402N"] = Network(
    name="JAMSTEC NIED DONET1 (broadband)",
    channels=132,
    starttime=datetime(2014, 10, 1, 0, 0),
    url=URL_Hinet,
)
NETWORK["0402A"] = Network(
    name="NIED DONET1 (strong motion)",
    channels=132,
    starttime=datetime(2014, 10, 1, 0, 0),
    url=URL_Hinet,
)
NETWORK["0402AN"] = Network(
    name="JAMSTEC NIED DONET1 (broadband)",
    channels=132,
    starttime=datetime(2014, 10, 1, 0, 0),
    url=URL_Hinet,
)

NETWORK["0501"] = Network(
    name="AIST",
    channels=81,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.aist.go.jp/index_en.html",
)

NETWORK["0601"] = Network(
    name="GSI",
    channels=6,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.gsi.go.jp/ENGLISH/index.html",
)

NETWORK["0701"] = Network(
    name="Tokyo Metropolitan Government",
    channels=54,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.metro.tokyo.jp/ENGLISH/index.htm",
)
NETWORK["0702"] = Network(
    name="Hot Spring Research Institute " "of Kanagawa Prefecture",
    channels=42,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.onken.odawara.kanagawa.jp/",
)
NETWORK["0703"] = Network(
    name="Aomori Prefectural Government",
    channels=15,
    starttime=datetime(2004, 4, 1, 0, 0),
    url="http://www.pref.aomori.lg.jp/foreigners/",
)
NETWORK["0705"] = Network(
    name="Shizuoka Prefectural Government",
    channels=3,
    starttime=datetime(2004, 6, 15, 0, 0),
    url="http://www.pref.shizuoka.jp/a_foreign/english/",
)

NETWORK["0801"] = Network(
    name="ADEP",
    channels=606,
    starttime=datetime(2015, 1, 1, 0, 0),
    url="http://www.adep.or.jp/",
)

NETWORK["010501"] = Network(
    name="NIED V-net (Tokachidake)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010502"] = Network(
    name="NIED V-net (Tarumaesan)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010503"] = Network(
    name="NIED V-net (Usuzan)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010504"] = Network(
    name="NIED V-net (Hokkaido-Komagatake)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010505"] = Network(
    name="NIED V-net (Iwatesan)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010506"] = Network(
    name="NIED V-net (Nasudake)",
    channels=9,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010507"] = Network(
    name="NIED V-net (Asamayama)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010508"] = Network(
    name="NIED V-net (Kusatsu-Shiranesan)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010509"] = Network(
    name="NIED V-net (Fujisan)",
    channels=57,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010510"] = Network(
    name="NIED V-net (Miyakejima)",
    channels=40,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010511"] = Network(
    name="NIED V-net (Izu-Oshima)",
    channels=39,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010512"] = Network(
    name="NIED V-net (Asosan)",
    channels=44,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010513"] = Network(
    name="NIED V-net (Unzendake)",
    channels=33,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)
NETWORK["010514"] = Network(
    name="NIED V-net (Kirishimayama)",
    channels=22,
    starttime=datetime(2010, 4, 1, 0, 0),
    url=URL_Vnet,
)

JMA_VNET = "JMA Volcanic Seismometer Network "

NETWORK["030201"] = Network(
    name=JMA_VNET + "(Atosanupuri)",
    channels=7,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030202"] = Network(
    name=JMA_VNET + "(Meakandake)",
    channels=16,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030203"] = Network(
    name=JMA_VNET + "(Taisetsuzan)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030204"] = Network(
    name=JMA_VNET + "(Tokachidake)",
    channels=23,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030205"] = Network(
    name=JMA_VNET + "(Tarumaesan)",
    channels=18,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030206"] = Network(
    name=JMA_VNET + "(Kuttara)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030207"] = Network(
    name=JMA_VNET + "(Usuzan)",
    channels=15,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030208"] = Network(
    name=JMA_VNET + "(Hokkaido-Komagatake)",
    channels=20,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030209"] = Network(
    name=JMA_VNET + "(Esan)",
    channels=7,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030210"] = Network(
    name=JMA_VNET + "(Iwakisan)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030211"] = Network(
    name=JMA_VNET + "(Akita-Yakeyama)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030212"] = Network(
    name=JMA_VNET + "(Iwatesan)",
    channels=10,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030213"] = Network(
    name=JMA_VNET + "(Akita-Komagatake)",
    channels=7,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030214"] = Network(
    name=JMA_VNET + "(Chokaisan)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030215"] = Network(
    name=JMA_VNET + "(Kurikomayama)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030216"] = Network(
    name=JMA_VNET + "(Zaozan)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030217"] = Network(
    name=JMA_VNET + "(Azumayama)",
    channels=17,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030218"] = Network(
    name=JMA_VNET + "(Adatarayama)",
    channels=8,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030219"] = Network(
    name=JMA_VNET + "(Bandaisan)",
    channels=11,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030220"] = Network(
    name=JMA_VNET + "(Nasudake)",
    channels=8,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030221"] = Network(
    name=JMA_VNET + "(Nikko-Shiranesan)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030222"] = Network(
    name=JMA_VNET + "(Kusatsu-Shiranesan)",
    channels=8,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030223"] = Network(
    name=JMA_VNET + "(Asamayama)",
    channels=24,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030224"] = Network(
    name=JMA_VNET + "(Niigata-Yakeyama)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030225"] = Network(
    name=JMA_VNET + "(Yakedake)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030226"] = Network(
    name=JMA_VNET + "(Norikuradake)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030227"] = Network(
    name=JMA_VNET + "(Ontakesan)",
    channels=8,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030228"] = Network(
    name=JMA_VNET + "(Hakusan)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030229"] = Network(
    name=JMA_VNET + "(Fujisan)",
    channels=18,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030230"] = Network(
    name=JMA_VNET + "(Hakoneyama)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030231"] = Network(
    name=JMA_VNET + "(Izu-Tobu Volcanoes)",
    channels=11,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030232"] = Network(
    name=JMA_VNET + "(Izu-Oshima)",
    channels=16,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030233"] = Network(
    name=JMA_VNET + "(Niijima)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030234"] = Network(
    name=JMA_VNET + "(Kozushima)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030235"] = Network(
    name=JMA_VNET + "(Miyakejima)",
    channels=14,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030236"] = Network(
    name=JMA_VNET + "(Hachijojima)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030237"] = Network(
    name=JMA_VNET + "(Aogashima)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030238"] = Network(
    name=JMA_VNET + "(Tsurumidake and Garandake)",
    channels=7,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030239"] = Network(
    name=JMA_VNET + "(Kujusan)",
    channels=4,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030240"] = Network(
    name=JMA_VNET + "(Asosan)",
    channels=18,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030241"] = Network(
    name=JMA_VNET + "(Unzendake)",
    channels=10,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030242"] = Network(
    name=JMA_VNET + "(Kirishimayama)",
    channels=19,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030243"] = Network(
    name=JMA_VNET + "(Sakurajima)",
    channels=25,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030244"] = Network(
    name=JMA_VNET + "(Satsuma-Iojima)",
    channels=7,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030245"] = Network(
    name=JMA_VNET + "(Kuchinoerabujima)",
    channels=13,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030246"] = Network(
    name=JMA_VNET + "(Suwanosejima)",
    channels=6,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
NETWORK["030247"] = Network(
    name=JMA_VNET + "(Hakkodasan)",
    channels=9,
    starttime=datetime(2010, 12, 1, 0, 0),
    url=URL_JMA,
)
